api/Dockerfile

```diff
@@ -1,14 +1,39 @@
-FROM node:18.15.0-alpine
+##################
+### BASE IMAGE ###
+##################
+FROM node:18.15.0-alpine AS base
+
+# Directory used in container
+WORKDIR /usr/api/
+
+# Copy dependency package, install, copy everything else
+COPY package*.json .
+RUN npm i --omit=dev
+COPY . .
+
+# Install TypeScript. Needed for build process.
+RUN npm i -D typescript
+
+# Compile to JavaScript build 
+RUN npm run build
+
+
+##################
+### PROD IMAGE ###
+##################
+FROM node:18.15.0-alpine as prod
 
 # Add curl for health check
 RUN apk --update --no-cache add curl
 
 # Directory used in container
 WORKDIR /usr/api/
 
-# Copy dependency package, install, copy everything else
+# Install packages. Needed even for compiled build.
 COPY package*.json .
```

Comment:
If you're running npm i instead of npm ci, you only need to copy package.json, and not package-lock.json

Reply:
Changed. Now doing all package installs in final image.